lib/rubocop/cop/rails/environment_variable_access.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for direct access to environment variables through the
+      # `ENV` variable within the application code. This can lead to runtime
+      # errors due to misconfiguration that could have been discovered at boot
+      # time if the environment variables were loaded as part of initialization
+      # and copied into the application's configuration or secrets. The cop can
+      # be configured to allow either reads or writes if required.
+      #
+      # @example
+      #   # bad
+      #   ENV["FOO"]
+      #   ENV.fetch("FOO")
+      #   ENV["FOO"] = "bar"
+      #
+      #   # good
+      #   Rails.application.config.foo
+      #   Rails.application.config.foo = "bar"
+      #   Rails.application.config.x.foo.bar
+      #   Rails.application.secrets.foo
```

Comment:
Thanks for adding the options. Can you update the examples?

```suggestion
      # @example
      #   # good
      #   Rails.application.config.foo
      #   Rails.application.config.x.foo.bar
      #   Rails.application.secrets.foo
      #   Rails.application.config.foo = "bar"
      #
      # @example AllowReads: false (default)
      #   # bad
      #   ENV["FOO"]
      #   ENV.fetch("FOO")
      #
      # @example AllowReads: true
      #   # good
      #   ENV["FOO"]
      #   ENV.fetch("FOO")
      #
      # @example AllowWrites: false (default)
      #   # bad
      #   ENV["FOO"] = "bar"
      #
      # @example AllowWrites: true
      #   # good
      #   ENV["FOO"] = "bar"
```